### DISCUSSION TITLE
Podman kube play building context

### DISCUSSION THREAD

**Discussion by rbrightling** [containers/podman#20602] [Q&A] [answered]
Hi,

I'm looking at moving from currently using compose to `podman kube` for basic local development environments. The setup is similar to [cookiecutter-django](https://github.com/cookiecutter/cookiecutter-django)

I'm having a small issue with getting `podman kube play --build kube.yaml` to build the images with context for the directory where the code is.

```
WORKDIR /code
COPY . /code
pip -r requirements.txt install
```

I can do `podman build --file python-app/Containerfile -t python-app .` works fine for building the images; but, would like to just use the kube command.

I've looked and found the `--context-dir` option for builds which enables me to move the Containerfile directories to different subdirectory; however, the context-dir of the image build itself seems to still be the Containerfile directory. `podman kube play --build --context-dir=$(pwd)/containers kube.yaml`

If I did a `podman run -it --rm python-app ls` it would only return "Contianerfile"... 

Is there an option CLI / kube YAML I'm missing which could help? or a possibility/interest of having this as a feature? 

cookiecutter-django local.yml Compose example:
```yaml
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
```

Thanks,

**Comment by rhatdan** [comment, marked as answer]
So you want the Containerfile to be located in a different location then the context dir.
Since Compose can do this, I think adding a --containerfile option (Aliased to --dockerfile) would be acceptable.

Interested in opening a PR?

**Reply by rbrightling** [reply]
Hi, 
Thanks,
(Sorry, Go is somewhat new to me; but, I had a look around). 
I think I'm looking to change buildOpts.ContextDirectory = filepath.Dir(buildFile) in podman/pkg/domain/infra/abi/play.go.

If I looked at adding a --build-context-dir to modify this or default to  filepath.Dir(buildFile) would that be good?

I can raise a feature request and give it a go.

**Reply by rhatdan** [reply]
Yes that seams resonable.  I would just have --context-dir and --containerfile as the two options.
